examples/render/src/main.rs

```diff
@@ -67,17 +66,137 @@ struct Args {
     #[arg(long = "height", default_value_t = 512)]
     height: u32,
 
-    /// Map mode (e.g. 'static', 'tile', 'continuous')
-    #[arg(short = 'm', long = "mode")]
-    mode: Option<String>,
+    /// Map mode
+    #[arg(short = 'm', long = "mode", default_value = "static")]
+    mode: Mode,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Default, clap::ValueEnum)]
+enum Mode {
+    #[default]
+    /// once-off still image of an arbitrary viewport
+    Static,
+    /// once-off still image of a single tile
+    Tile,
+    /// continually updating map
+    Continuous,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, clap::ValueEnum)]
+enum DebugMode {
+    /// Edges of tile boundaries are shown as thick, red lines to help diagnose tile clipping issues.
+    TileBorders,
+    ParseStatus,
+    /// Each tile shows a timestamp indicating when it was loaded.
+    Timestamps,
+    /// Edges of glyphs and symbols are shown as faint, green lines to help diagnose collision and label placement issues.
+    Collision,
+    /// Each drawing operation is replaced by a translucent fill. Overlapping drawing operations appear more prominent to help diagnose overdrawing.
+    Overdraw,
+    StencilClip,
+    DepthBuffer,
+}
+
+impl From<DebugMode> for MapDebugOptions {
+    fn from(value: DebugMode) -> Self {
+        match value {
+            DebugMode::TileBorders => MapDebugOptions::TileBorders,
+            DebugMode::ParseStatus => MapDebugOptions::ParseStatus,
+            DebugMode::Timestamps => MapDebugOptions::Timestamps,
+            DebugMode::Collision => MapDebugOptions::Collision,
+            DebugMode::Overdraw => MapDebugOptions::Overdraw,
+            DebugMode::StencilClip => MapDebugOptions::StencilClip,
+            DebugMode::DepthBuffer => MapDebugOptions::DepthBuffer,
+        }
+    }
+}
+
+impl Args {
+    fn render(self) -> Vec<u8> {
+        let mut map = ImageRendererOptions::new();
+        map.with_api_key(self.apikey.unwrap_or_default());
+        map.with_cache_path(self.cache.to_string_lossy().to_string());
+        map.with_asset_root(self.asset_root.to_string_lossy().to_string());
+        map.with_pixel_ratio(self.ratio);
+        map.with_size(self.width, self.height);
+
+        match self.mode {
+            Mode::Static => {
+                let mut map = map.build_static_renderer();
+                if let Some(debug) = self.debug {
+                    map.set_debug_flags(debug.into());
+                }
+                map.set_style_url(&self.style);
+                map.set_camera(
+                    self.x as f64,
+                    self.y as f64,
```

Comment:
Current `maplibre_native`-API mixes lat/lon and tiles..
Seems like a footgun for our users whith not good solution.

We should likely expose a different enum representing XYZ which has two constructors => `from_lat_lon(zoom:f64,lat:f64,lon:f64)` and `from_xy(z:u8,x:32,y:32)` and can convert between them.

Reply:
I haven't even looked at the args here yet -- they were ported from the MLN example. Going forward, we should start with a simple `u8/u32/u32`  z/x/y. I don't want to support floats just yet ... maybe later.